map/apply-scheduled-updates-to-preplanned-map-area/src/main/java/com/esri/samples/apply_scheduled_updates_to_preplanned_map_area/ApplyScheduledUpdatesToPreplannedMapAreaSample.java

```diff
@@ -0,0 +1,252 @@
+                      offlineMapSyncParameters.setRollbackOnFailure(true);
+
+                      // create a sync job using the parameters
+                      OfflineMapSyncJob offlineMapSyncJob = offlineMapSyncTask.syncOfflineMap(offlineMapSyncParameters);
+
+                      // start the job and get the results
+                      offlineMapSyncJob.start();
+                      offlineMapSyncJob.addJobDoneListener(() -> {
+                        if (offlineMapSyncJob.getStatus() == Job.Status.SUCCEEDED) {
+                          OfflineMapSyncResult offlineMapSyncResult = offlineMapSyncJob.getResult();
+
+                          // if mobile map package reopen is required, close the existing mobile map package and load it again
+                          if (offlineMapSyncResult.isMobileMapPackageReopenRequired()) {
+                            mobileMapPackage.close();
+                            mobileMapPackage.loadAsync();
+                            mobileMapPackage.addDoneLoadingListener(() -> {
+                              if (mobileMapPackage.getLoadStatus() == LoadStatus.LOADED && !mobileMapPackage.getMaps().isEmpty()) {
+
+                                // add the map from the mobile map package to the map view
+                                mapView.setMap(mobileMapPackage.getMaps().get(0));
+
+                              } else {
+                                new Alert(Alert.AlertType.ERROR, "Failed to load the mobile map package.").show();
+                              }
+                            });
+                          }
+
+                          // update labels
+                          updateAvailableLabel.setText("Updates: Up to date");
+                          updateSizeLabel.setText("Update size: N/A");
+
+                        } else {
+                          new Alert(Alert.AlertType.ERROR, "Error syncing the offline map: " + offlineMapSyncJob.getError().getMessage()).show();
+                        }
+
+                        // disable the 'Apply Scheduled Updates' button
+                        applyUpdatesButton.setDisable(true);
+                        // hide progress indicator
+                        progressIndicator.setVisible(false);
+
+                      });
+                    } catch (InterruptedException | ExecutionException ex) {
+                      new Alert(Alert.AlertType.ERROR, "Error creating DefaultOfflineMapSyncParameters" + ex.getMessage()).show();
+                    }
+                  });
+                });
+
+              } else {
+                updateAvailableLabel.setText("Updates: NOT AVAILABLE");
+              }
+            } catch (Exception e) {
+              new Alert(Alert.AlertType.ERROR, "Error checking for Scheduled Updates Availability: " + e.getMessage()).show();
+            }
+          });
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Failed to load the mobile map package.").show();
```

Comment:
Could get the message from the mobile map package here like you do for your other errors.
`mobileMapPackage.getLoadError().getMessage()`

Usefully, it will report a file not found in case users haven't provisioned correctly or the data is corrupted etc.

Reply:
We generally don't do this in java since the content of the alert is not considered useful for the end-user. I suppose devs can print these to the stack trace to debug their apps if they need?